projects/vdk-core/src/vdk/internal/builtin_plugins/config/log_config.py

```diff
@@ -81,13 +82,19 @@ def configure_initial_logging_before_anything():
     3. When initialize_job hook (applicable for vdk run only) is executed then below configure_loggers function
         is run which adds more context to the logs and initializes syslog handler (if configured to do so)
     """
-    log_level = "WARNING"
-    if os.environ.get(LOG_LEVEL_VDK, None):
-        log_level = os.environ.get(LOG_LEVEL_VDK)
-    elif os.environ.get("VDK_LOG_LEVEL_VDK", None):
-        log_level = os.environ.get("VDK_LOG_LEVEL_VDK")
-
-    logging.basicConfig(format="%(message)s", level=logging.getLevelName(log_level))
+    warnings.warn(
+        "The vdk-core logging configuration is deprecated and will be removed in a future version. "
+        "Please use vdk-structlog for logging configuration.",
+        DeprecationWarning,
+    )
+    if not os.environ.get("VDK_USE_STRUCTLOG"):
+        log_level = "WARNING"
+        if os.environ.get(LOG_LEVEL_VDK, None):
+            log_level = os.environ.get(LOG_LEVEL_VDK)
+        elif os.environ.get("VDK_LOG_LEVEL_VDK", None):
+            log_level = os.environ.get("VDK_LOG_LEVEL_VDK")
+
+        logging.basicConfig(format="%(message)s", level=logging.getLevelName(log_level))
 
 
 def configure_loggers(
```

Comment:
You should add the check in log_config.py where configure_loggers is called as well no?

Reply:
Added the check as part of the PR #2944.